src/lib/Components/Home/ArtistRails/ArtistRail.tsx

```diff
@@ -177,30 +177,26 @@ export class ArtistRail extends Component<Props, State> {
   }
 
   refreshData = () => {
-    // TODO: Ensures rail has been mounted before a refresh occurs, circumventing setState errors.
-    // See https://stackoverflow.com/a/40969739/1038901 for more info.
-    if (this.refs.rail) {
```

Comment:
This guard would never pass, because `refreshData` is called from `componentDidMount` and that’s apparently called before the `render` that would add the `ref`.

Reply:
@damassi I’d like to know steps to reproduce the warning you were trying to fix, I haven’t seen it occur yet.

Reply:
(But it was definitely only a warning, so this won’t lead to any runtime crashes.)

Reply:
@alloy - if you comment out the ref check the issue should happen -- And it _appears_ as though its a warning and things dont break but its actually a more serious error in that from this point on no more rail refreshing occurs. (Meaning, if 10 out of 14 rails have been refreshed and the error occurs on the 11th iteration, 12, 13, and 14 will contain stale data.)

> This guard would never pass, because refreshData is called from componentDidMount and that’s apparently called before the render that would add the ref.

This is incorrect -- `componentDidMount` is the last thing that executes, post render. And since a ref is effectively a "unit" of DOM it _will_ execute since the dom is there and mounted. See https://reactjs.org/docs/react-component.html#componentdidmount. 

I took care to make sure that everything was working like it should so I'm wondering what the issue was exactly or if perhaps there was some kind of regression or change? Because it is super cool if we can remove the `ref` hack, but note that the warning _does_ stop rail fetch execution.

Reply:
I read the docs the same way, but think there may be a slight difference in React Native, as there is no DOM in the same way those docs refer to. That’s not to say that RN maybe should work closer to that, but it was not doing so in my testing.

Reply:
Hmm 🤔 I just dropped in some logs to test and unless theres something odd about this `componentDidMount` is definitely firing last: 

<img width="213" alt="screen shot 2017-12-21 at 11 54 30 am" src="https://user-images.githubusercontent.com/236943/34272620-c525f884-e645-11e7-9035-50669db5e6c9.png">

Reply:
Weird that this even worked, `ref` in react native is a function: https://github.com/artsy/emission/blob/13535321a2fd30efda15fadf9a137ba2fbf4bf78/src/lib/Components/Consignments/Screens/Metadata.tsx#L151

Reply:
@orta - They can be strings or functions or references. 

For bug, see https://github.com/artsy/emission/pull/910

Reply:
I didn’t look further into the reason for the artist rail not having a `ref` on mounting, because I solved the actual problem in a different way.